dotcom-rendering/src/web/components/SecureSignup.tsx

```diff
@@ -0,0 +1,163 @@
+// https://developers.google.com/recaptcha/docs/faq#id-like-to-hide-the-recaptcha-badge.-what-is-allowed
+
+type Props = { newsletterId: string; successText: string };
+
+const termsStyle = css`
+	${textSans.xxsmall()}
+	color: ${text.supporting};
+	a {
+		${textSans.xxsmall()}
+		text-decoration: none;
+		:hover {
+			text-decoration: underline;
+		}
+	}
+	strong {
+		color: ${neutral[0]};
+		font-weight: bold;
+	}
+`;
+
+const PrivacyTerms = () => {
+	return (
+		<span css={termsStyle}>
+			<strong>Privacy Notice: </strong>
+			Newsletters may contain info about charities, online ads, and
+			content funded by outside parties. For more information see our{' '}
+			<Link
+				href="https://www.theguardian.com/help/privacy-policy"
+				rel="noopener noreferrer"
+			>
+				privacy policy
+			</Link>
+			.&nbsp;
+		</span>
+	);
+};
+
+const RecaptchaTerms = () => (
+	<span css={termsStyle}>
+		We operate Google reCaptcha to protect our website and the Google{' '}
+		<Link
+			href="https://policies.google.com/privacy"
+			rel="noopener noreferrer"
+		>
+			Privacy Policy
+		</Link>{' '}
+		and{' '}
+		<Link
+			href="https://policies.google.com/terms"
+			rel="noopener noreferrer"
+		>
+			Terms of Service
+		</Link>{' '}
+		apply.
+	</span>
+);
+
+const generateForm = (
+	newsletterId: string,
+): { html: string; styles: string } => {
+	const cache = createCache({ key: 'email-signup-iframe' });
+
+	// eslint-disable-next-line @typescript-eslint/unbound-method -- not a react method, 'this' binding not required?
+	const { extractCriticalToChunks, constructStyleTagsFromChunks } =
+		createEmotionServer(cache);
+
+	const html = renderToString(
+		<CacheProvider value={cache}>
+			<form id={`secure-signup-${newsletterId}`}>
+				<Label text="Enter your email address" />
+
+				<div
+					css={css`
+						display: flex;
+						flex-direction: row;
+						align-items: flex-end;
+						flex-wrap: wrap;
+					`}
+				>
+					<div
+						css={css`
+							margin-right: ${space[3]}px;
+							flex-basis: 335px;
+							flex-shrink: 1;
+						`}
+					>
+						<TextInput
+							hideLabel={true}
+							name="email"
+							label="Enter your email address"
+							type="email"
+						/>
+					</div>
+					<Button
+						cssOverrides={css`
+							justify-content: center;
+							background-color: ${neutral[0]};
+							:hover {
+								background-color: ${neutral[20]};
+							}
+							flex-basis: 118px;
+							flex-shrink: 0;
+							margin-top: ${space[2]}px;
+						`}
+						type="submit"
+					>
+						Sign up
+					</Button>
+				</div>
+			</form>
+		</CacheProvider>,
+	);
+	const chunks = extractCriticalToChunks(html);
+	const styles = constructStyleTagsFromChunks(chunks);
+
+	return { html, styles };
+};
+
+export const SecureSignup = ({ newsletterId, successText }: Props) => {
+	const { html, styles } = generateForm(newsletterId);
+
+	return (
+		<>
+			<Island
+				clientOnly={true}
+				deferUntil={'idle'}
```

Comment:
Are we able to use 'visible' here?

Reply:
I think given the results suggested in this PR: https://github.com/guardian/dotcom-rendering/pull/5276 it would be better to keep this as on idle as you have it!